src/haz3lweb/view/ScratchMode.re

```diff
@@ -98,7 +98,180 @@ let download_slide_state = state => {
   let json_data = ScratchSlide.export(state);
   JsUtil.download_json("hazel-scratchpad", json_data);
 };
+let breadcrumb_bar =
+    (
+      ~inject,
+      ~model as
+        {
+          editors,
+          //font_metrics,
+          //show_backpack_targets,
+          settings,
+          //mousedown,
+          //langDocMessages,
+          _,
+          //results,
+        }: Model.t,
+    ) => {
+  let editor = Editors.get_editor(editors);
+  let zipper = editor.state.zipper;
+  let unselected = Zipper.unselect_and_zip(zipper);
+  let (term, _) = MakeTerm.go(unselected);
+  let info_map = Statics.mk_map(term);
+  switch (zipper.backpack, Indicated.index(zipper)) {
+  | ([_, ..._], _) => [div([text("")])]
+  | (_, None) => [div([text("")])]
+  | (_, Some(id)) =>
+    switch (Id.Map.find_opt(id, info_map)) {
+    | None => [div([text("")])]
+    | Some(ci) =>
+      let ancestors = Info.ancestors_of(ci);
+      let ancestors =
+        List.map(
+          a => {
+            switch (Id.Map.find_opt(a, info_map)) {
+            | Some(v) =>
+              switch (v) {
+              | Info.InfoExp({term, ancestors, _}) =>
+                switch (term.term) {
+                | Fun(_) => List.hd(ancestors)
+                | _ => Uuidm.nil
+                }
+              | _ => Uuidm.nil
+              }
+            | None => Uuidm.nil
+            }
+          },
+          ancestors,
+        );
+      let view_fun = (name, ids, level) => [
+        div(
+          ~attr=
+            Attr.many([
+              clss(["breadcrumb_bar_function"]),
+              Attr.on_click(_ =>
+                inject(Update.Set(Breadcrumb_bar(level, true)))
+              ),
+              Attr.on_click(_ =>
+                inject(
+                  UpdateAction.PerformAction(Jump(TileId(List.hd(ids)))),
+                )
+              ),
+            ]),
+          [text(name)],
+        ),
+      ];
+      //if (List.exists(a => a == List.hd(ids), ancestors)) {
+      //} else {
+      //  [];
+      //};
+      let funs_display = Array.make(10, []);
```

Comment:
Generally in ReasonML we try to avoid using arrays, and use lists instead - it's because adding "mutability" makes it easy for errors to creep in - not that any mutability errors have crept in here, but just for a general rule.

Reply:
Fixed